pkg/tasks/security/authorization/trust_domain.go

```diff
@@ -27,64 +28,143 @@ import (
 func cleanupTrustDomainMigration() {
 	util.Log.Info("Cleanup")
 	util.KubeDeleteContents("foo", TrustDomainPolicy)
-	sleep := examples.Sleep{"foo"}
-	httpbin := examples.Httpbin{"foo"}
+	sleep := examples.Sleep{Namespace: "foo"}
+	httpbin := examples.Httpbin{Namespace: "foo"}
 	sleep.Uninstall()
 	httpbin.Uninstall()
-	sleep = examples.Sleep{"bar"}
+	sleep = examples.Sleep{Namespace: "bar"}
 	sleep.Uninstall()
-	util.Shell(`kubectl patch -n istio-system smcp/basic --type merge -p '{"spec":{"security":{"trust":{"domain":"cluster.local"}}}}'`)
-	util.Shell(`kubectl patch -n istio-system smcp/basic --type merge -p '{"spec":{"security":{"dataPlane":{"mtls":false},"controlPlane":{"mtls":false}}}}'`)
-	util.Shell(`oc -n istio-system wait --for condition=Ready smcp/basic --timeout 180s`)
-	time.Sleep(time.Duration(40) * time.Second)
+	applyTrustDomain("cluster.local", "")
 }
 
 func TestTrustDomainMigration(t *testing.T) {
 	defer cleanupTrustDomainMigration()
 	defer util.RecoverPanic(t)
 
 	util.Log.Info("Trust Domain Migration")
-	util.Log.Info("Enable Control Plane MTLS")
-	util.Shell(`kubectl patch -n istio-system smcp/basic --type merge -p '{"spec":{"security":{"dataPlane":{"mtls":true},"controlPlane":{"mtls":true}}}}'`)
```

Comment:
@jwendell Should we enable control plane mtls in this trust domain test case ? By default, the SMCP mtls spec are false.

Reply:
Why? The upstream docs don't do that.

Reply:
The upstream doc mentioned install Istio and mutual TLS enabled
https://istio.io/v1.9/docs/tasks/security/authorization/authz-td-migration/#before-you-begin

Reply:
I'm not seeing the mTLS being enabled anywhere in those instructions. The demo profile does not enable mTLS. I believe this is an error in the docs. I've asked upstream folks about this.

Meanwhile, I've enabled it in this test. It doesn't affect the test behavior anyway.

Reply:
I just got a clarification from upstream. That doc is indeed outdated, since mTLS is installed by default.

My confusion was about the PERMISSIVE vs STRICT mTLS policy. This feature does not require STRICT to work. And, by changing `spec.security.dataPlane.mtls: true` actually sets the policy from PERMISSIVE to STRICT.

So, it's not strictly necessary. That said, given I already added it, I don't think it hurts, so, let's leave it there.

Reply:
Thanks for checking that. I am clear now.